contracts/RewardsLiquidator.vy

```diff
@@ -147,98 +239,156 @@ def _get_eth_anchor_price() -> uint256:
+    decimal_token_in: uint256,
+    decimal_token_out: uint256
+) -> uint256:
+    amount_out: uint256 = (amount * price) / (10 ** decimal_token_in) # = (amount * (10 ** (18 - decimal_token_in)) * price) / 10 ** 18
     min_mult: uint256 = 10**18 - max_diff_percent
-    return (amount_out * min_mult) / 10**18
-
+    return (amount_out * min_mult) / (10 ** (36 - decimal_token_out)) # = ((amount_out * min_mult) / 10**18) / (10 ** (18 - decimal_token_out))
 
+# 1) stETH -> ETH (Curve)
+# 2) ETH -> USDC (Uniswap v3)
+# 3) USDC -> UST (Curve)
 @external
 def liquidate(ust_recipient: address) -> uint256:
     assert msg.sender == self.vault, "unauthorized"
 
     steth_amount: uint256 = ERC20(STETH_TOKEN).balanceOf(self)
     assert steth_amount > 0, "zero stETH balance"
 
-    steth_anchor_price: uint256 = self._get_steth_anchor_price()
-
+    # steth -> eth
+    steth_eth_price: uint256 = self._get_chainlink_price(CHAINLINK_STETH_ETH_FEED, False)
     min_eth_amount: uint256 = self._get_min_amount_out(
         steth_amount,
-        steth_anchor_price,
-        self.max_steth_price_difference_percent
+        steth_eth_price,
+        self.max_steth_eth_price_difference_percent,
+        STETH_TOKEN_DECIMALS,
+        WETH_TOKEN_DECIMALS
     )
 
-    ERC20(STETH_TOKEN).approve(CURVE_STETH_POOL, steth_amount)
+    ERC20(STETH_TOKEN).approve(self.curve_steth_pool, steth_amount)
 
-    eth_amount: uint256 = CurveStableSwap(CURVE_STETH_POOL).exchange(
-        CURVE_STETH_INDEX,
-        CURVE_ETH_INDEX,
+    eth_amount: uint256 = CurvePool(self.curve_steth_pool).exchange(
+        self.curve_steth_index,
+        self.curve_eth_index,
         steth_amount,
         min_eth_amount
     )
-
     assert eth_amount >= min_eth_amount, "insuff. ETH return"
-    assert self.balance >= eth_amount
-
-    eth_anchor_price: uint256 = self._get_eth_anchor_price()
+    assert self.balance >= eth_amount, "ETH balance mismatch"
 
-    min_ust_amount: uint256 = self._get_min_amount_out(
+    # eth -> usdc
+    eth_usdc_price: uint256 = self._get_chainlink_price(CHAINLINK_USDC_ETH_FEED, True)
+    min_usdc_amount: uint256 = self._get_min_amount_out(
         eth_amount,
-        eth_anchor_price,
-        self.max_eth_price_difference_percent
+        eth_usdc_price,
+        self.max_eth_usdc_price_difference_percent,
+        WETH_TOKEN_DECIMALS,
+        USDC_TOKEN_DECIMALS
     )
 
-    ust_amount_actual: uint256 = self._sushi_sell_eth_to_ust(
+    usdc_amount: uint256 = self._uniswap_v3_sell_eth_to_usdc(
         eth_amount,
-        min_ust_amount,
-        ust_recipient
+        min_usdc_amount,
+        self
     )
 
-    assert ust_amount_actual >= min_ust_amount, "insuff. UST return"
+    assert usdc_amount >= min_usdc_amount, "insuff. USDC return"
+    assert ERC20(USDC_TOKEN).balanceOf(self) >= usdc_amount, "USDC balance mismatch"
+
+    # usdc -> ust
+    usdc_eth_price: uint256 = self._get_chainlink_price(CHAINLINK_USDC_ETH_FEED, False)
```

Comment:
Could we do this without querying Chainlink the second time (the first call at line 327)? E.g. by adding a `_get_inverse_rate` function, caching the `USDC/ETH` rate, and using it both on this line and line 327 (passing it to `_get_inverse_rate`)?

Reply:
I'd even remove the `inverse` parameter from the `_get_chainlink_price` function and use `_get_inverse_rate` every time an inverse rate is needed, but it's a matter of code style/preference.

Reply:
good point, fixed